README.md
# decorativeAppBar

A Flutter package to easily decorate your appbar with rounded borders and adding your own widgets on appbar.

## Features

1)Compatible with any widget class of flutter
2)Compatible with both android and ios
3)Has an additional feature of styling your appbar gradient,size,background,bottom radius.

## Usage
in pubspec.yaml ```decorative_app_bar: ^0.0.2```
Import it as ```import 'package:decorative_app_bar/decorative_app_bar.dart'```

## Snippet For Reference
```
DecorativeAppBar(
          barHeight: 400,  //height of appbar
          barPad: 170,     //distance of the curve from appbar
          radii: 100,      //radius of the curve
          background: Colors.pink[50],   //background color must be same as app's background
          gradient1: Colors.white, //gradient color for appbar
          gradient2: Colors.green, //gradient color for appbar
          extra: extra_widget(),  //style your icons,searchbar,text to your appbar with your own widget
 ));
```
## Screenshots

#### default
![image](https://raw.githubusercontent.com/Ruparna97/appbar_example_images/master/images/default.jpeg)

#### feature example 1
![image](https://raw.githubusercontent.com/Ruparna97/appbar_example_images/master/images/adding_features2.jpeg)

#### feature example 2
![image](https://raw.githubusercontent.com/Ruparna97/appbar_example_images/master/images/adding_features1.jpeg)

CHANGELOG.md
## [0.0.1] - 2020-09-24

* Testing

## [0.0.2] - 2020-09-24

* Adding explanations and example


test/decorativeAppBar_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:decorative_app_bar/decorativeAppBar.dart';

void main() {}


lib/decorativeAppBar.dart
library decorative_app_bar;

import 'package:flutter/material.dart';

// class DecorativeAppBar extends StatefulWidget {
//   final double barHeight, barPad, radii;
//   final Color gradient1, gradient2, background;
//   final Widget extra;

//   DecorativeAppBar(
//       {Key key,
//       @required this.barHeight,
//       @required this.barPad,
//       @required this.radii,
//       this.background = Colors.white,
//       this.gradient1 = Colors.white,
//       this.gradient2 = const Color(0xffffccbc),
//       this.extra})
//       : super(key: key);
//   @override
//   _DecorativeAppBarState createState() => _DecorativeAppBarState();
// }

// class _DecorativeAppBarState extends State<DecorativeAppBar> {
//   @override
//   Widget build(BuildContext context) {
//     double z = widget.barHeight - widget.barPad;
//     return Container(
//         height: widget.barHeight,
//         decoration: BoxDecoration(
//           gradient:
//               LinearGradient(colors: [widget.gradient1, widget.gradient2]),
//         ),
//         child: Stack(children: [
//           if (widget.extra != null) widget.extra,
//           Column(
//             children: [
//               Container(
//                 height: z,
//                 margin: EdgeInsets.only(top: widget.barPad),
//                 decoration: BoxDecoration(
//                   color: widget.background,
//                   borderRadius:
//                       BorderRadius.vertical(top: Radius.circular(widget.radii)),
//                   boxShadow: [
//                     BoxShadow(
//                       color: Colors.grey.withOpacity(0.2),
//                       blurRadius: 5,
//                       spreadRadius: 0.2,
//                       offset: Offset(0.5, -8), // changes position of shadow
//                     ),
//                   ],
//                 ),
//               )
//             ],
//           ),
//         ]));
//   }
// }

// class DecorativeAppBar {
//   appBar(
//       {@required double barHeight,
//       @required double barPad,
//       @required double radii,
//       Color background = Colors.white,
//       Color gradient1 = Colors.white,
//       Color gradient2 = const Color(0xffffccbc),
//       Widget extra}) async {
//     double z = barHeight - barPad;
//     return Container(
//         height: barHeight,
//         decoration: BoxDecoration(
//           gradient: LinearGradient(colors: [gradient1, gradient2]),
//         ),
//         child: Stack(children: [
//           if (extra != null) extra,
//           Column(
//             children: [
//               Container(
//                 height: z,
//                 margin: EdgeInsets.only(top: barPad),
//                 decoration: BoxDecoration(
//                   color: background,
//                   borderRadius:
//                       BorderRadius.vertical(top: Radius.circular(radii)),
//                   boxShadow: [
//                     BoxShadow(
//                       color: Colors.grey.withOpacity(0.2),
//                       blurRadius: 5,
//                       spreadRadius: 0.2,
//                       offset: Offset(0.5, -8), // changes position of shadow
//                     ),
//                   ],
//                 ),
//               )
//             ],
//           ),
//         ]));
//   }
// }

class DecorativeAppBar extends StatelessWidget {
  final double barHeight, barPad, radii;
  final Color gradient1, gradient2, background;
  final Widget extra;

  DecorativeAppBar(
      {Key key,
      @required this.barHeight,
      @required this.barPad,
      @required this.radii,
      this.background = Colors.white,
      this.gradient1 = Colors.white,
      this.gradient2 = const Color(0xffffccbc),
      this.extra})
      : super(key: key);
  @override
  Widget build(BuildContext context) {
    double z = barHeight - barPad;
    return Container(
        height: barHeight,
        decoration: BoxDecoration(
          gradient: LinearGradient(colors: [gradient1, gradient2]),
        ),
        child: Stack(children: [
          if (extra != null) extra,
          Column(
            children: [
              Container(
                height: z,
                margin: EdgeInsets.only(top: barPad),
                decoration: BoxDecoration(
                  color: background,
                  borderRadius:
                      BorderRadius.vertical(top: Radius.circular(radii)),
                  boxShadow: [
                    BoxShadow(
                      color: Colors.grey.withOpacity(0.2),
                      blurRadius: 5,
                      spreadRadius: 0.2,
                      offset: Offset(0.5, -8), // changes position of shadow
                    ),
                  ],
                ),
              )
            ],
          ),
        ]));
  }
}


lib/decorative_app_bar.dart
library decorative_app_bar;

import 'package:flutter/material.dart';

class DecorativeAppBar extends StatefulWidget {
  final double barHeight, barPad, radii;
  final Color gradient1, gradient2, background;
  final Widget extra;

  DecorativeAppBar(
      {Key key,
      @required this.barHeight,
      @required this.barPad,
      @required this.radii,
      this.background = Colors.white,
      this.gradient1 = Colors.white,
      this.gradient2 = const Color(0xffffccbc),
      this.extra})
      : super(key: key);
  @override
  _DecorativeAppBarState createState() => _DecorativeAppBarState();
}

class _DecorativeAppBarState extends State<DecorativeAppBar> {
  @override
  Widget build(BuildContext context) {
    double z = widget.barHeight - widget.barPad;
    return Container(
        height: widget.barHeight,
        decoration: BoxDecoration(
          gradient:
              LinearGradient(colors: [widget.gradient1, widget.gradient2]),
        ),
        child: Stack(children: [
          if (widget.extra != null) widget.extra,
          Column(
            children: [
              Container(
                height: z,
                margin: EdgeInsets.only(top: widget.barPad),
                decoration: BoxDecoration(
                  color: widget.background,
                  borderRadius:
                      BorderRadius.vertical(top: Radius.circular(widget.radii)),
                  boxShadow: [
                    BoxShadow(
                      color: Colors.grey.withOpacity(0.2),
                      blurRadius: 5,
                      spreadRadius: 0.2,
                      offset: Offset(0.5, -8), // changes position of shadow
                    ),
                  ],
                ),
              )
            ],
          ),
        ]));
  }
}


example/example.dart
import 'package:flutter/material.dart';
import 'package:decorative_app_bar/decorative_app_bar.dart';

class Example extends StatefulWidget {
  @override
  _ExampleState createState() => _ExampleState();
}

class _ExampleState extends State<Example> {
  final Shader linearGradient = LinearGradient(
    colors: <Color>[Colors.white, Color(0xffffccbc)],
  ).createShader(Rect.fromLTWH(0.0, 0.0, 200.0, 70.0));

  Widget text() {
    return Container(
      child: Row(mainAxisAlignment: MainAxisAlignment.spaceBetween, children: [
        Padding(
          padding: EdgeInsets.only(top: 50, left: 20),
          child: Text("Hey There,",
              style: TextStyle(
                letterSpacing: 4.0,
                fontSize: 40,
              )),
        ),
        Padding(
            padding: EdgeInsets.only(top: 45, right: 20),
            child: Icon(
              Icons.notifications,
              color: Colors.black,
              size: 40.0,
            )),
      ]),
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        backgroundColor: Colors.pink[50],
        body: DecorativeAppBar(
          barHeight: 400,
          barPad: 170,
          radii: 100,
          background: Colors.pink[50],
          gradient1: Colors.white,
          gradient2: Colors.green,
          extra: text(),
        ));
  }
}
